EngineLayer/Proteomics/PeptideWithSetModifications.cs

```diff
@@ -190,7 +201,7 @@ public PeptideWithSetModifications Localize(int j, double massToLocalize)
             }
 
             vvv.Add(j + 2, new ModificationWithMass(null, null, null, TerminusLocalization.Any, massToLocalize + massOfExistingMod, null, new List<double> { 0 }, new List<double> { massToLocalize + massOfExistingMod }, null));
-            var hm = new PeptideWithSetModifications(modPep, vvv, numFixedMods);
+            var hm = new PeptideWithSetModifications(vvv, numFixedMods, this, OneBasedStartResidueInProtein, OneBasedEndResidueInProtein);
```

Comment:
For NonSpecific search modes, you had previously requested a removal of modPep from newly generated PWSM. When modPep is null, modPep.Protein throws a null pointer exception. The new constructor generates a modPep.Protein by taking this.Protein instead.